tools/migration/db/run.sh

```diff
@@ -1,125 +1,505 @@
+    fi
+
+    get_version
+    if [ "$cur_version" = "$target_version" ]; then
+        echo "It has always running the $target_version, no longer need to upgrade."
+        exit 0
+    fi
+
+    set +e
+    version_com $cur_version '1.5.0'
+    v1_com=$?
+    version_com $target_version '1.5.0'
+    v2_com=$?
+    ## if no version specific, see it as larger then 1.5.0
+    if [ "$target_version" = 'head' ]; then
+        v2_com=1
+    fi
+    set -e
+
+    # $cur_version <='1.5.0', $target_version <='1.5.0', it needs to call mysql upgrade.
+    if [ $v1_com != 1 ] && [ $v2_com != 1 ]; then
+        if [ $ISMYSQL != true ]; then
+            echo "Please make sure to mount the correct the data volumn."
+            return 1
+        else
+            alembic_up mysql $target_version
+            return $?
+        fi
+    fi
+
+    # $cur_version <='1.5.0', $target_version >'1.5.0', it needs to upgrade to 1.5.0.mysql => 1.5.0.pgsql => target_version.pgsql.
+    if [ $v1_com != 1 ] && [ $v2_com = 1 ]; then
+        if [ $ISMYSQL != true ]; then
+            echo "Please make sure to mount the correct the data volumn."
+            return 1
+        else
+            alembic_up mysql '1.5.0' false
+
+            ## dump mysql
+            mysqldump --compatible=postgresql --default-character-set=utf8 --databases registry > /harbor-migration/db/registry.mysql     
+            stop_mysql $DB_USR $DB_PWD
+
+            ## migrate 1.5.0-mysql to 1.5.0-pqsql.
+            python /harbor-migration/db/pgsql_migrator.py /harbor-migration/db/registry.mysql /harbor-migration/db/registry.pgsql
+
+            launch_pgsql $PGSQL_USR
+            psql -U $PGSQL_USR -f /harbor-migration/db/schema/registry_from_$cur_version.pgsql
+            psql -U $PGSQL_USR -f /harbor-migration/db/schema/notaryserver.pgsql
+            psql -U $PGSQL_USR -f /harbor-migration/db/schema/notarysigner.pgsql
+            psql -U $PGSQL_USR -f /harbor-migration/db/registry.pgsql
+
+            ## it needs to call the alembic_up to target, disable it as it's now unsupported.
+            #alembic_up $target_version
+            stop_pgsql
+
+            rm -rf /var/lib/mysql/*
+            cp -rf $PGDATA/* /var/lib/mysql
+            return 0
+        fi        
+    fi
+
+    # $cur_version > '1.5.0', $target_version > '1.5.0', it needs to pgsql upgrade.    
+    if [ $v1_com = 1 ] && [ $v2_com = 1 ]; then
+        if [ $ISPGSQL != true ]; then
+            echo "Please make sure to mount the correct the data volumn."
+            return 1
+        else
+            alembic_up pgsql $target_version
+            return $?
+        fi
+    fi
+
+    echo "Unsupported DB upgrade from $cur_version to $target_version, please check the inputs."
+    return 1
+}
+
+# This function is only for notary <= 1.5.0 mysql to pgsql.
+function up_notary {
+
+    set +e
+    chown -R 10000:10000 /var/lib/mysql
+    mysqld &
+    for i in {60..0}; do
```

Comment:
Could we update `launch_mysql` to make it support different scenarios and pass an empty string as password?